src/mscorlib/src/System/Span.cs

```diff
@@ -612,7 +612,13 @@ internal static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
         {
             if (byteLength == 0)
                 return;
-
+            
+#if AMD64
+            if (byteLength > uint.MaxValue) goto PInvoke;
```

Comment:
Good point. So for 4000 bytes length, we see 17% improvement. For 8000 bytes length, we see 7% improvement. Considering GC, how about setting threshold as 4096 bytes?